sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -590,6 +591,23 @@ object TypeCoercion {
   }
 
   /**
+   * Coerces NullTypes of a Stack function to the corresponding column types.
+   */
+  object StackCoercion extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan resolveExpressions {
+      case s @ Stack(children) if s.childrenResolved && s.children.head.dataType == IntegerType &&
```

Comment:
we can put `s.children.head.dataType == IntegerType && s.children.head.foldable` in `Stack` as a method